src/doc/blend_funcs.cpp

```diff
@@ -409,9 +446,15 @@ static void set_sat(double& r, double& g, double& b, double s)
   ((x) > (y) ? ((y) > (z) ? (y) : ((x) > (z) ? (z) : (x))) :                                       \
                ((y) > (z) ? ((z) > (x) ? (z) : (x)) : (y)))
 
-  double& min = MIN(r, MIN(g, b));
-  double& mid = MID(r, g, b);
-  double& max = MAX(r, MAX(g, b));
+  // Fetch channel indices
+  const uint8_t imin = get_imin_channel(r, g, b);
+  const uint8_t imax = get_imax_channel(r, g, b);
+  const uint8_t imid = get_imid_channel(imin, imax);
+
+  // Map the indices for each channel to references
+  double& min = index_to_ref(r, g, b, imin);
+  double& max = index_to_ref(r, g, b, imax);
+  double& mid = index_to_ref(r, g, b, imid);
```

Comment:
Just wanted to suggest the alternative implementation I was talking about
```suggestion
  // Order components from min to max
  if (r > g)
    std::swap(r, g);
  if (g > b)
    std::swap(g, b);
  if (r > g)
    std::swap(r, g);

  double& min = r;
  double& mid = g;
  double& max = b;
```

Reply:
I had to use pointers instead of references for the ordering to work, but it's working nicely. If rewriting the function to use pointers is no problem, then I'd go with this one for readability.

To briefly touch on performance: it varies from platform to platform. This alternative is likely the more performant approach on x86_64 when compiling with clang; glancing at the disassembly I'm seeing that it's generating much better code (branchless and around half as many instructions), but for other platforms I couldn't tell.

Reply:
I'll be testing the solution of this PR as it is in e359cb70105fa05da7bbf1af28445f5f447a1851 and trying to merge it.

In case someone wants to compare performance of different solutions, `src/doc/blend_benchmark.cpp` is a good place to start.